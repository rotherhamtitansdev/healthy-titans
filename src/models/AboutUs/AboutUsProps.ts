export interface AboutUsProps {
  section_1: string[];
  section_2: string[];
  section_3: string[];
  section_4: string[];
}