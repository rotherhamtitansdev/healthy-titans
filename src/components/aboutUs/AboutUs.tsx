// empty div at bottom for layout issue
/* eslint-disable react/self-closing-comp */
import { DocumentData } from "firebase/firestore";
import React, { useEffect, useState } from "react";
import FirebaseAPI from "../../api/FirebaseAPI";
import { AboutUsProps } from "../../models/AboutUs/AboutUsProps";
import AppHeader from "../app_header/AppHeader";
import MenuHeader from "../app_header/header/MenuHeader";
import RTLogo from "../../img/RT_Logo.png";

const gridStyles = "grid grid-cols-1 md:grid-cols-2";
const sectionPaddingStyles = "py-6 md:py-14";
const sectionHeaderStyles = "font-semibold text-[#252948] text-xl md:text-2xl lg:text-4xl py-5";
const imgContainerStyles = "flex justify-center items-center my-6 md:my-0";
const textMarginStyles = "mx-8 md:mx-16 lg:mx-24";
const textAreaStyles = `${textMarginStyles} flex flex-col gap-4`;

const AboutUs = () => {
  const [AboutUsData, setAboutUsData] = useState<AboutUsProps | undefined>();
  const [imgUrl, setImgUrl] = useState<string | undefined>("");

  useEffect(() => {
    FirebaseAPI.fetchAboutUsData().then((res) => {
      if (res) {
        setAboutUsData(res as DocumentData as AboutUsProps);
      }
    });

    FirebaseAPI.fetchImages("FoodCategories/Fruits/FruitAndVeg.svg").then((value) =>
      setImgUrl(value)
    );
  }, []);

  return (
    <>
      <AppHeader>
        <MenuHeader
          title="About Us"
          body={"Here's a bit about this app, who we are and why we've done it."}
        />
      </AppHeader>
      {AboutUsData && (
        <div className="font-semibold text-titansDarkGrey text-sm md:text-base lg:text-xl">
          <div id="section1" className={`${gridStyles} ${sectionPaddingStyles}`}>
            <div id="textArea1" className={textAreaStyles}>
              <h2 className={sectionHeaderStyles}>{AboutUsData.section_1[0]}</h2>
              <p>{AboutUsData.section_1[1]}</p>
              <p>{AboutUsData.section_1[2]}</p>
            </div>
            <div id="imgArea1" className={imgContainerStyles}>
              <img
                src={imgUrl}
                alt="fruit and veg"
                className="h-[200px] w-[200px] md:h-[300px] md:w-[300px] lg:h-[400px] lg:w-[400px]"
              />
            </div>
          </div>
          <div id="section2" className={`${gridStyles} ${sectionPaddingStyles} bg-white`}>
            <div id="imgArea2" className={`${imgContainerStyles} order-last md:order-first`}>
              <img
                src={RTLogo}
                alt="Rotherham Titans Logo"
                className="h-[265px] w-[228px] lg:h-[388px] lg:w-[331px]"
              />
            </div>
            <div id="textArea2" className={textAreaStyles}>
              <h2 className={sectionHeaderStyles}>{AboutUsData.section_2[0]}</h2>
              <p>{AboutUsData.section_2[1]}</p>
              <p>{AboutUsData.section_2[2]}</p>
            </div>
          </div>
          <div id="section3" className={sectionPaddingStyles}>
            <h2 className={`${sectionHeaderStyles} ${textMarginStyles} md:max-w-[40%]`}>
              {AboutUsData.section_3[0]}
            </h2>
            <div className={gridStyles}>
              <p className={`${textMarginStyles}  mb-4 md:mb-0`}>{AboutUsData.section_3[1]}</p>
              <p className={textMarginStyles}>{AboutUsData.section_3[2]}</p>
            </div>
          </div>
          <div id="section4" className={`${sectionPaddingStyles} bg-white`}>
            <h2 className={`${sectionHeaderStyles} ${textMarginStyles}`}>
              {AboutUsData.section_4[0]}
            </h2>
            <div className={gridStyles}>
              <div className={`${textAreaStyles} mb-4 md:mb-0`}>
                <p>{AboutUsData.section_4[1]}</p>
                <p>{AboutUsData.section_4[2]}</p>
              </div>
              <div className={textAreaStyles}>
                <p>{AboutUsData.section_4[3]}</p>
                <p>{AboutUsData.section_4[4]}</p>
              </div>
            </div>
          </div>
          <div className="h-[50px] bg-gradient-to-b from-white to-transparent"></div>
        </div>
      )}
    </>
  );
};
export default AboutUs;
