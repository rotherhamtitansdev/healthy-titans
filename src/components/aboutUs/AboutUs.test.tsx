import React from "react";
import { render, screen } from "@testing-library/react";
// import * as FirebaseAPI from "../../api/FirebaseAPI";
import AboutUs from "./AboutUs";

const mockContent = {
  section_1: [
    "section 1 header",
    "section 1 body",
    "section 1 body 2",
    "section 1 body 3",
    "section 1 body 4",
  ],
  section_2: ["section 2 header", "section 2 body", "section 2 body 2"],
  section_3: ["section 3 header", "section 3 body"],
  section_4: ["section 4 header", "section 4 body"],
  section_5: ["section 4 header", "section 4 body", "section 4 body 2"],
};

describe("About Us", () => {
  test("component renders with sections and image", async () => {
    jest.mock("../../api/FirebaseAPI", () => ({
      ...jest.requireActual("../../api/FirebaseAPI"),
      fetchImages: jest.fn().mockReturnValue("fruit.jpg"),
      fetchDataFromSubpath: jest.fn().mockReturnValue(mockContent),
    }));

    // jest.spyOn(FirebaseAPI, "fetchDataFromSubpath").mockResolvedValueOnce(mockContent);
    // jest.spyOn(FirebaseAPI, "fetchImages").mockResolvedValue("fruit.jpg");

    const { asFragment } = render(<AboutUs />);

    expect(await screen.findByText("section 1 header")).toBeVisible();
    expect(await screen.findByRole("img", { name: "green hearts" })).toHaveAttribute(
      "src",
      "fruit.jpg"
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
