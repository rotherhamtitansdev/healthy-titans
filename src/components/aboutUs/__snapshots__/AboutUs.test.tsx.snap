// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Us component renders with sections and image 1`] = `
<DocumentFragment>
  <div
    class="bg-mobileWave sm:bg-wave bg-no-repeat pb-2 bg-span"
  >
    <div
      class="flex flex-wrap h-[22rem]"
    >
      <div
        class="mx-auto pt-[3rem] md:pt-[2rem]"
      >
        <h1
          class="text-titansDarkBlue font-bold text-center sm:mt-[1rem] text-3xl sm:text-5xl lg:text-6xl mt-8 pb-4"
          data-testid="header-title"
        >
          About Us
        </h1>
        <div
          class="font-medium text-homepageHeaderSubTitle text-sm md:text-base lg:text-xl text-center w-2/3 lg:w-[45rem] mx-auto"
        >
          Here's a bit about this app, who we are and why we've done it.
        </div>
      </div>
    </div>
  </div>
  <div
    class="font-semibold text-titansDarkGrey text-sm md:text-base lg:text-xl"
  >
    <div
      class="grid grid-cols-1 md:grid-cols-2 py-6 md:py-14"
      id="section1"
    >
      <div
        class="mx-8 md:mx-16 lg:mx-24 flex flex-col gap-4"
        id="textArea1"
      >
        <h2
          class="font-semibold text-titansDarkBlue text-xl md:text-2xl lg:text-4xl py-5"
        >
          section 1 header
        </h2>
        <p>
          section 1 body
        </p>
        <p>
          section 1 body 2
        </p>
      </div>
      <div
        class="mx-8 md:mx-16 lg:mx-24 flex flex-col gap-4"
        id="imgArea1"
      >
        <img
          alt="speechmark"
          class="max-w-[62px] lg:max-w-[75px] py-5"
          src="fruit.jpg"
        />
        <p
          class="font-semibold text-xl lg:text-2xl text-titansDarkBlue"
        >
          section 1 body 3
        </p>
        <p
          class="text-base lg:text-xl font-medium"
        >
          section 1 body 4
        </p>
      </div>
    </div>
    <div
      class="grid grid-cols-1 md:grid-cols-2 py-6 md:py-14 bg-white"
      id="section2"
    >
      <div
        class="flex justify-center items-center my-6 md:my-0 order-last md:order-first"
        id="imgArea2"
      >
        <img
          alt="Titans Community Foundation Logo"
          class="h-[265px] w-[228px] lg:h-[388px] lg:w-[331px]"
          src="TitansCommunityLogo.png"
        />
      </div>
      <div
        class="mx-8 md:mx-16 lg:mx-24 flex flex-col gap-4"
        id="textArea2"
      >
        <h2
          class="font-semibold text-titansDarkBlue text-xl md:text-2xl lg:text-4xl py-5"
        >
          section 2 header
        </h2>
        <p>
          section 2 body
        </p>
        <p>
          section 2 body 2
        </p>
      </div>
    </div>
    <div
      class="grid grid-cols-1 md:grid-cols-2 py-6 md:py-14"
      id="section3"
    >
      <div
        class="mx-8 md:mx-16 lg:mx-24 flex flex-col gap-4"
        id="textArea3"
      >
        <h2
          class="font-semibold text-titansDarkBlue text-xl md:text-2xl lg:text-4xl py-5"
        >
          section 3 header
        </h2>
        <p>
          section 3 body
        </p>
      </div>
      <div
        class="flex justify-center items-center my-6 md:my-0 hidden md:flex"
        id="imgArea3"
      >
        <img
          alt="green hearts"
          class="hidden md:flex md:h-[268px] md:w-[268px]"
          src="example.jpg"
        />
      </div>
    </div>
    <div
      class="grid grid-cols-1 md:grid-cols-2 py-6 md:py-14 bg-white"
      id="section4"
    >
      <div
        class="flex justify-center items-center my-6 md:my-0 hidden md:flex"
        id="imgArea4"
      >
        <img
          alt="1 in 4 people"
          class="lg:h-[315px] lg:w-[315px] md:h-[170px] md:w-[170px]"
          src="logo.png"
        />
      </div>
      <div
        class="mx-8 md:mx-16 lg:mx-24 flex flex-col gap-4"
        id="textArea4"
      >
        <h2
          class="font-semibold text-titansDarkBlue text-xl md:text-2xl lg:text-4xl py-5"
        >
          section 4 header
        </h2>
        <p>
          section 4 body
        </p>
      </div>
    </div>
    <div
      class="grid grid-cols-1 md:grid-cols-2 py-6 md:py-14"
      id="section5"
    >
      <div
        class="mx-8 md:mx-16 lg:mx-24 flex flex-col gap-4"
        id="textArea5"
      >
        <h2
          class="font-semibold text-titansDarkBlue text-xl md:text-2xl lg:text-4xl py-5"
        >
          section 4 header
        </h2>
        <p>
          section 4 body
        </p>
        <p>
          section 4 body 2
        </p>
      </div>
      <div
        class="flex justify-center items-center my-6 md:my-0 order-first md:order-last"
        id="imgArea5"
      >
        <img
          alt="Cookie's Khaw Logo"
          class="w-[270px] lg:w-[453px]"
          src="Logo.svg"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
